Rename `continue_on` and `start_on` to `continues_on` and `starts_on` with deprecation warnings

msimberg opened the pull request:
Fix #1410.

Note that I've named the `__starts_on` namespace `__starts_on_ns` because the former would conflict with the `__starts_on` concept (https://github.com/NVIDIA/stdexec/blob/fbd384d1bcbe9dfb9805765f7f40e8dff417a662/include/stdexec/__detail/__transform_sender.hpp#L239). Let me know if you'd prefer it the other way around.

ericniebler commented on examples/nvexec/maxwell_distributed.cpp:
```diff
@@ -478,13 +478,13 @@ int main(int argc, char *argv[]) {
     auto compute_h = ex::when_all(
-      ex::just() | exec::on(gpu, ex::bulk(bulk_cells, bulk_h_update)),
-      ex::just() | exec::on(gpu_with_priority, ex::bulk(border_cells, border_h_update))
+      ex::just() | exec::starts_on(gpu, ex::bulk(bulk_cells, bulk_h_update)),
+      ex::just() | exec::starts_on(gpu_with_priority, ex::bulk(border_cells, border_h_update))
```
code that uses `exec::on` (as opposed to `stdexec::on`) is expecting there-and-back-again behavior, so it should not be changed to `starts_on`. here and elsewhere.

msimberg replied:
Thanks, good catch. I've removed these (and related) changes in the last batch of changes.

ericniebler commented on README.md:
```diff
@@ -41,9 +41,9 @@ int main()
-        stdexec::on(sched, stdexec::just(0) | stdexec::then(fun)),
-        stdexec::on(sched, stdexec::just(1) | stdexec::then(fun)),
-        stdexec::on(sched, stdexec::just(2) | stdexec::then(fun))
+        stdexec::starts_on(sched, stdexec::just(0) | stdexec::then(fun)),
+        stdexec::starts_on(sched, stdexec::just(1) | stdexec::then(fun)),
+        stdexec::starts_on(sched, stdexec::just(2) | stdexec::then(fun))
```
:+1:

ericniebler commented on include/stdexec/__detail/__execution_fwd.hpp:
```diff
@@ -194,26 +194,36 @@ namespace stdexec {
-  extern const transfer_t transfer;
+  using continue_t [[deprecated("continue_on_t has been renamed continues_on_t")]] = continues_on_t;
+  [[deprecated("continue_on has been renamed continues_on")]]
+  extern const continues_on_t continue_on;
```
thanks for the deprecation warnings. :clap:

ericniebler commented on include/stdexec/__detail/__on.hpp:
```diff
@@ -171,9 +171,9 @@ namespace stdexec {
+                continues_on(                                                            //
                   __forward_like<_Data>(__clsur)(                                       //
-                    continue_on(                                                        //
+                    continues_on(                                                        //
```
pls run clang-format over this pr.

msimberg replied:
Done. Which version of clang-format do you expect? With 18 I get formatting changes on unrelated parts of the codebase (which I've left out of the changes here), so I was wondering if you expect a specific version. clang-format tends to change its mind about formatting between versions...

ericniebler replied:
stdexec has been formatted with clang-format-17. i intend to reformat everything with clang-format-18 as a separate change.

ericniebler commented on test/stdexec/algos/adaptors/test_let_error.cpp:
```diff
@@ -341,7 +341,7 @@ namespace {
   }
 
-  // Return a different sender when we invoke this custom defined on implementation
+  // Return a different sender when we invoke this custom defined let_error implementation
```
nice catch

ericniebler reviewed (commented):
love this. one comment on `exec::on` but otherwise looks good.

ericniebler commented:
/ok to test

msimberg commented:
I pushed some more changes that I hope addresses your comments. I missed some renamings in the nvexec directory (I don't have nvhpc set up on my stdexec build at the moment) the first time around. I hope those are all good now as well.

ericniebler commented:
/ok to test
